server/sonar-server/src/main/java/org/sonar/server/rule/RegisterRules.java

```diff
@@ -130,6 +130,8 @@ public void start() {
       keysToIndex.addAll(removedRules.stream().map(RuleDefinitionDto::getKey).collect(Collectors.toList()));
 
       persistRepositories(dbSession, context.repositories());
+      // FIXME lack of resiliency, active rules index is corrupted if rule index fails
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Take the required action to fix the issue indicated by this comment. ([squid:S1134](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fsb%2FSONAR-10052%2Fqprofiles&issues=AWE-qvQxmckxG3vIth7-&open=AWE-qvQxmckxG3vIth7-)

[ISSUE_KEY]: AWE-qvQxmckxG3vIth7-